financial-connections/src/main/java/com/stripe/android/financialconnections/features/common/ModalBottomSheetContent.kt

```diff
@@ -78,198 +95,213 @@ internal fun LegalDetailsBottomSheetContent(
-                    text = bullet.content,
-                    defaultStyle = v3Typography.bodySmall.copy(
-                        color = v3Colors.textSubdued
-                    ),
-                    onClickableTextClick = onClickableTextClick
-                )
+            AnnotatedText(
+                text = firstText,
+                defaultStyle = titleStyle.copy(color = v3Colors.textDefault),
+                onClickableTextClick = onClickableTextClick
+            )
+            secondText?.let {
+                Row {
+                    AnnotatedText(
+                        text = requireNotNull(bullet.content),
+                        defaultStyle = v3Typography.bodySmall.copy(color = v3Colors.textSubdued),
+                        onClickableTextClick = onClickableTextClick
+                    )
+                }
             }
         }
     }
 }
 
 @Composable
-private fun BulletIcon(icon: ImageResource?) {
+private fun ListItemIcon(icon: ImageResource?) {
+    val bulletColor = v3Colors.iconDefault
+    val iconSize = 20.dp
     val modifier = Modifier
-        .size(20.dp)
-        .offset(y = 2.dp)
-    if (icon == null) {
-        val color = v3Colors.iconDefault
-        Canvas(
-            modifier = Modifier
-                .size(20.dp)
-                .padding(6.dp)
-                .offset(y = 2.dp),
-            onDraw = { drawCircle(color = color) }
+        .size(iconSize)
+        .offset(y = 1.dp)
+    when (icon) {
+        // Render a bullet if no icon is provided
+        null -> Canvas(
+            modifier = modifier.padding((iconSize - 8.dp) / 2),
+            onDraw = { drawCircle(color = bulletColor) }
         )
-    } else {
-        when (icon) {
-            is ImageResource.Local -> Image(
-                modifier = modifier,
-                painter = painterResource(id = icon.resId),
-                contentDescription = null,
-            )
 
-            is ImageResource.Network -> StripeImage(
-                url = icon.url,
-                errorContent = {
-                    val color = v3Colors.iconDefault
-                    Canvas(
-                        modifier = Modifier
-                            .size(6.dp)
-                            .align(Alignment.Center),
-                        onDraw = { drawCircle(color = color) }
-                    )
-                },
-                imageLoader = LocalImageLoader.current,
-                contentDescription = null,
-                modifier = modifier
-            )
-        }
+        // Render the icon if it's a local resource
+        is ImageResource.Local -> Image(
+            modifier = modifier,
+            painter = painterResource(id = icon.resId),
+            contentDescription = null,
+        )
+
+        // Render the icon if it's a network resource, or fallback to a bullet if it fails to load
+        is ImageResource.Network -> StripeImage(
+            url = icon.url,
```

Comment:
Noticing this in the screen recording: Should we provide a placeholder shape in `StripeImage` if it’s trying to load a URL?

Reply:
that's a great idea! actually realized adding a shimmer placeholder + using `AnimatedContent` to transition between loading placeholder -> success / error image content is a nicer approach that what I had! - https://github.com/stripe/stripe-android/pull/7634/commits/2631f18e661e248982ad4cff89f422ab97e6bc3e